src/tqec/plaquette/plaquette.py

```diff
@@ -26,15 +32,23 @@ def __init__(self, qubits: list[PlaquetteQubit], circuit: ScheduledCircuit) -> N
             coordinate system.
         :param circuit: scheduled quantum circuit implementing the computation that the
             plaquette should represent.
+        :param origin: origin of the plaquette. This is used to compute the global position
+            during the generation of the circuit.
+        :param origin_index: index of the origin qubit in the list. Only used in combination
+            with origin == PlaquetteOrigin.USER_DEFINED.
+
+        :raises ValueError: if the number of qubits doesn't match the number of qubits
         """
+        # if len(qubits) != len(circuit.raw_circuit.all_qubits()):
+        #     raise ValueError("Number of qubits doesn't match number of circuit qubits")
+
         self._qubits = qubits
         self._circuit = circuit
+        self._caclulate_origin(origin, origin_index)
 
     @property
-    def shape(self) -> Shape2D:
-        maxx = max(qubit.position.x for qubit in self.qubits)
-        maxy = max(qubit.position.y for qubit in self.qubits)
-        return Shape2D(maxx + 1, maxy + 1)
+    def origin(self) -> Position:
+        return self._origin
```

Comment:
Following another comment, if the origin is implicitly defined by the user input (i.e., always (0, 0)), this method may be replaced by
```suggestion
    def origin(self) -> Position:
        return Position(0, 0)
```